tedana/utils.py

```diff
@@ -181,12 +184,20 @@ def dice(arr1, arr2):
     if arr1.shape != arr2.shape:
         raise ValueError('Shape mismatch: arr1 and arr2 must have the same shape.')
 
-    arr_sum = arr1.sum() + arr2.sum()
-    if arr_sum == 0:
-        dsi = 0
+    if axis is not None and axis > (arr1.ndim - 1):
+        raise ValueError('Axis provided {} not supported by the input arrays.'.format(axis))
+
+    if axis is None:
+        arr1 = arr1.flatten()
+        arr2 = arr2.flatten()
+        axis = 0
```

Comment:
This check isn't necessary; if `axis` is `None` then any calls to `.sum(axis=None)` will implicitly flatten the array.

Reply:
Thanks for catching.